### DISCUSSION TITLE
How to properly add json-ld script?

### DISCUSSION THREAD

**Discussion by kiranimeid** [nuxt/nuxt#16864] [Questions] [answered]
hi, currently trying nuxt-3, but when i try to add json-ld to my page, it's not working properly.
here is what i've tried:

**1. using useMeta()**
```
useMeta({
	script: [
		{
			type: 'application/ld-json',
			json: jsonLd,
		},
	],
});
```

result: `<script type="application/ld-json" json="[object Object]"></script>`

**2. using `<Script>` tag**
```
<Script type="application/ld-json" v-html="jsonLd"></Script>
```
result: `<script type="application/ld-json" innerhtml="[object Object]"></script>`
or
```
<Script type="application/ld-json">{{jsonLd}}</Script>
```
result: `<script type="application/ld-json"></script>`

am i doing something wrong? thanks.

**Comment by viii47** [comment, marked as answer]
Hey @kiranimeid,

You need to define your content in a `children` reference.

### useMeta Example
```js
    useMeta({
      script: [
        {
          hid: 'json-ld-reference-bla',
          children: '{ "@context": "https://schema.org", "@type": "Organization", "name": "yes" }',
          type: 'application/ld+json'
        }
      ]
    })
```
### Template Example
For template-typing, you need to use `children=""`

```js
<Script type="application/ld-json" :children="jsonLd"></Script>
```

**Reply by kiranimeid** [reply]
ah, i dont know about the children. Thank you, now it's working. :)

**Reply by viii47** [reply]
🥳 - yeah I think `vue/usehead()` uses `children` generally to inherit a sub-attribute which is great, otherwise you'd always need to know the specific property value

**Comment by Marvin1003** [comment]
Does Nuxt 3 patch the built-in vue component `<component />` to not allow `is="script"`? I have a framework agnostic component-library that needs to render inline scripts of `type="application/ld+json"` and when consumed in a SSR environment it should be able to render the components with structured data on the server. `<component is="script" />` is a hacky workaround but it works in Vue 2 & 3. Nuxt 2 also still renders it but encoded. Nuxt 3 doesn't render it at all, `<ClientOnly />` and ` ssr: false` also don't change this behavior. Is there a way to get this working in Nuxt 2 & 3 or can you recommend a different approach that is framework agnostic and supports Vue 2 & 3? 
@pi0 maybe you can give any insight why this differs from the vue standard.
